anchore_engine/services/policy_engine/engine/vulns/mappers.py

```diff
@@ -16,38 +16,452 @@
+         "version": "8.3"
+        }
+        """
+        distro = distro_dict.get("type")
+        if distro.lower() == UbuntuMapper.grype_os:
+            distro = UbuntuMapper.engine_distro
+        elif distro.lower() == DebianMapper.grype_os:
+            distro = DebianMapper.engine_distro
+        else:
+            raise ValueError(
+                "Expected distro to be {} or {} but found {}".format(
+                    UbuntuMapper.grype_os, DebianMapper.grype_os, distro
+                )
+            )
+
+        version = distro_dict.get("version")
+
+        return "{}:{}".format(distro, version)
+
+
+class ApkgMapper(PackageMapper):
+    engine_type = "APKG"
+    grype_type = "apk"
+
+    def to_grype_artifact(
+        self,
+        image_package: ImagePackage,
+        location_cpes_dict: Dict[str, List[str]] = None,
+    ):
+        artifact = super().to_grype_artifact(image_package, location_cpes_dict)
+        # TODO
+        # if image_package.normalized_src_pkg != "N/A":
+        #     artifact["metadata"] = {"sourceRpm": image_package.normalized_src_pkg}
+        return artifact
+
+    def get_feed_group_from_grype_match_distro(self, distro_dict):
+        """
+        {
+         "type": "redhat",
+         "version": "8.3"
+        }
+        """
+        distro = distro_dict.get("type")
+        if distro.lower() == AlpineMapper.grype_os:
+            distro = AlpineMapper.engine_distro
+        else:
+            raise ValueError(
+                "Expected distro to be {} but found {}".format(
+                    AlpineMapper.grype_os, distro
+                )
+            )
+
+        version = distro_dict.get("version").split(".", 1)[0]
+
+        return "{}:{}".format(distro, version)
+
+
+class CPEMapper(PackageMapper):
+    engine_type = None
+    grype_type = None
+
+    def to_grype_artifact(
+        self,
+        image_package: ImagePackage,
+        location_cpes_dict: Dict[str, List[str]] = None,
+    ):
+        artifact = super().to_grype_artifact(image_package)
+        artifact["cpes"] = [
+            # cpe : 2.3 : part : vendor : product : version : update : edition : language : sw_edition : target_sw : target_hw : other
+            "cpe:2.3:{}".format(
+                ":".join(
+                    [
+                        item.cpetype,  # part
+                        item.vendor,  # vendor
+                        item.name,  # product
+                        item.version,  # version
+                        item.update,  # update
+                        item.meta,  # edition
+                        "-",  # language
+                        "-",  # sw_edition
+                        "-",  # target_sw
+                        "-",  # target_hw
+                        "-",  # other
+                    ]
+                )
+            )
+            for item in location_cpes_dict.get(image_package.pkg_path)
+        ]
+
+        return artifact
+
+
+class PythonMapper(CPEMapper):
```

Comment:
Same comment here. These all seem to be the same class, just with different input values. And I'm not clear on why the to_grype_artifact() method is overridden in each subclass since it looks to always be the same and always a call to super() for the same method.